python/ray/services.py

```diff
@@ -1276,502 +1266,248 @@ def start_raylet_monitor(redis_address,
+            following parameters will be set to default values if it's None:
+            node_ip_address("127.0.0.1"), num_local_schedulers(1),
+            include_log_monitor(True), worker_path(path of default_worker.py)
         cleanup (bool): If cleanup is true, then the processes started here
             will be killed by services.cleanup() when the Python process that
             called this method exits.
-        redirect_worker_output: True if the stdout and stderr of worker
-            processes should be redirected to files.
-        redirect_output (bool): True if stdout and stderr for non-worker
-            processes should be redirected to files and false otherwise.
-        include_log_monitor (bool): If True, then start a log monitor to
-            monitor the log files for all processes on this node and push their
-            contents to Redis.
-        include_webui (bool): If True, then attempt to start the web UI. Note
-            that this is only possible with Python 3.
-        start_workers_from_local_scheduler (bool): If this flag is True, then
-            start the initial workers from the local scheduler. Else, start
-            them from Python.
-        resources: A dictionary mapping resource name to the quantity of that
-            resource.
-        plasma_directory: A directory where the Plasma memory mapped files will
-            be created.
-        huge_pages: Boolean flag indicating whether to start the Object
-            Store with hugetlbfs support. Requires plasma_directory.
-        autoscaling_config: path to autoscaling config file.
-        plasma_store_socket_name (str): If provided, it will specify the socket
-            name used by the plasma store.
-        raylet_socket_name (str): If provided, it will specify the socket path
-            used by the raylet process.
-        temp_dir (str): If provided, it will specify the root temporary
-            directory for the Ray process.
-        _internal_config (str): JSON configuration for overriding
-            RayConfig defaults. For testing purposes ONLY.
 
     Returns:
         A dictionary of the address information for the processes that were
             started.
     """
 
-    set_temp_root(temp_dir)
+    set_temp_root(ray_params.temp_dir)
 
     logger.info("Process STDOUT and STDERR is being redirected to {}.".format(
         get_logs_dir_path()))
 
-    config = json.loads(_internal_config) if _internal_config else None
-
-    if resources is None:
-        resources = {}
-    if not isinstance(resources, list):
-        resources = num_local_schedulers * [resources]
+    config = json.loads(
+        ray_params._internal_config) if ray_params._internal_config else None
 
-    if num_workers is not None:
+    ray_params.apply_when_none(
+        include_log_monitor=True,
+        resources={},
+        num_local_schedulers=1,
+        node_ip_address="127.0.0.1")
```

Comment:
Instead of calling `apply_when_none` here, maybe it's better to just set default values in the `RayParams` class?

Reply:
The `None` parameters are sometimes used as flags. If we assign default values to them, there could be some test failures.